### ISSUE TITLE
smolvla_robocasa is systematically underevaluated: eval instructions are out-of-distribution, and the docs' rename_map contradicts the checkpoint's embedded preprocessor

### ISSUE THREAD

**Issue by cengishasan** [huggingface/lerobot#4548]
## Summary

While evaluating `lerobot/smolvla_robocasa` on RoboCasa365 I found two independent
evaluation-protocol problems. Together they make `lerobot-eval` report ~0% success for a
checkpoint that demonstrably performs well above that when evaluated in its training regime.

1. **Instruction distribution mismatch (the big one).** The checkpoint was trained on
   `lerobot/robocasa_target_human_unified`, whose task strings (`meta/tasks.parquet`) are the 50
   **bare task names** (`OpenDrawer`, `CloseFridge`, …). But `lerobot.envs.robocasa.RoboCasaEnv`
   feeds the policy **natural-language instructions** from `get_ep_meta()["lang"]`
   ("Open the right drawer."). The policy has never seen a sentence at train time.
   Measured effect (identical env seeds, device=mps):

   | Instruction | Result |
   |---|---|
   | natural language (current eval default) | **0/41** episodes across 13 atomic `target` tasks |
   | bare task name (training distribution)  | **4/9** on a 3-task subset of the same seeds — CloseFridge **3/3**, OpenDrawer **1/3**, TurnOnSinkFaucet 0/3 |

   Qualitatively, language-conditioned rollouts still produce smooth, scene-appropriate motion
   (approach the right region, no flailing), which makes the failure easy to misattribute to
   "weak policy" rather than OOD conditioning.

2. **Docs' `--rename_map` example contradicts the checkpoint.** `docs/source/robocasa.mdx`
   suggests:
   `left→camera1, eye_in_hand→camera2, right→camera3`,
   but the checkpoint's own `policy_preprocessor.json` (training-time pipeline) stores:
   `left→camera1, right→camera2, eye_in_hand→camera3`.
   I verified against the training data itself: the dataset's video content under each native
   key matches the env render under the same key (two shoulder views + wrist view), so the
   embedded map is the correct one and the docs example swaps two cameras. Note also that
   `lerobot-eval` passes `cfg.rename_map` as a `preprocessor_overrides` entry, so a user
   following the docs replaces the correct stored map with the wrong one.

## Environment

- lerobot 0.6.1 (PyPI), torch 2.11.0, Python 3.12.7 (arm64), macOS 15 (M1 Max), `--policy.device=mps`
- robocasa v1.0.1 (commit `a07e365`), robosuite master (commit `5ce6643`), mujoco 3.3.1, numpy 2.2.5
- checkpoint `lerobot/smolvla_robocasa` @ `38d38bf58e1217092c5a81418fcc8eef1a7c3a6f`
- object registry: lightwheel only (per the docs' asset download instructions)

## Reproduction

Baseline (current behavior, natural language):

```
lerobot-eval \
  --policy.path=lerobot/smolvla_robocasa \
  --env.type=robocasa --env.task=CloseFridge --env.split=target \
  --eval.batch_size=1 --eval.n_episodes=20 --eval.use_async_envs=false \
  --policy.device=mps
```

→ ~0% success (in my runs: 0/3 on seeds 500–502; 0/41 across 13 atomic tasks).

Training-distribution instructions (no CLI flag exists; monkeypatch used):

```python
from lerobot.envs import robocasa as rc
_orig = rc.RoboCasaEnv.reset
def reset_name(self, *a, **kw):
    out = _orig(self, *a, **kw)
    self.task_description = self.task   # bare name, as in meta/tasks.parquet
    return out
rc.RoboCasaEnv.reset = reset_name
```

→ CloseFridge 3/3, OpenDrawer 1/3 on the same seeds.

Checking the two maps:

```python
import json
from huggingface_hub import hf_hub_download
p = hf_hub_download("lerobot/smolvla_robocasa", "policy_preprocessor.json")
print(json.load(open(p))["steps"][0]["config"]["rename_map"])
# right→camera2, eye_in_hand→camera3  (docs say the opposite)
```

## Suggested fixes

1. Either (a) add an env/eval option to feed the dataset-consistent task string (and make it the
   documented default for this checkpoint), or (b) regenerate/fine-tune the checkpoint with the
   `lang` instructions so the natural-language eval is in-distribution. As-is, every published
   or private eval of this checkpoint through `lerobot-eval` underreports it.
2. Fix the `rename_map` example in `docs/source/robocasa.mdx` to match the embedded
   preprocessor — or drop it from the example entirely and let the stored pipeline apply, and
   make sure an absent `--rename_map` does not override the stored map with an empty one.
3. (Minor) Consider logging a warning when eval-time task strings have zero overlap with the
   training dataset's task vocabulary — it would have made this a 5-minute diagnosis.

Happy to provide the full per-episode JSONL logs or a PR for any of the above.

**Comment by sattyamjjain** [comment]
This looks like an instruction-distribution artifact rather than a policy failure. If the eval task strings are not the strings the checkpoint was trained against, a near-zero score is what you would expect, and it says nothing about the policy.

The cheap way to tell the two apart: run the same suite twice, once with the training-distribution instruction strings and once with yours, same seed, same episodes. If the first arm scores and the second does not, it is the strings. If both floor, it is the checkpoint or the environment wiring.

I run matched-pair evals like this on VLA policies, and the control arm is routinely the thing that turns a confusing number into an obvious one. Happy to share the runner if useful; it is Apache-2.0.

**Comment by atirna** [comment]
the rename map is a separate mismatch here. The checkpoint embeds left → camera1, right → camera2, eye_in_hand → camera3, while the docs/CI command had right and eye_in_hand swapped. #4578 makes the command match that stored mapping.

the instruction-string control still looks useful for the eval question. If you can share the runner, I can use it to reproduce both arms with the same task set and episodes.

**Comment by sattyamjjain** [comment]
Thanks, the rename map explains the wiring half. The instruction-string control is the other half, and it is still worth running.

Here is the runner. It is the same matched-pair setup I use on VLA policies: two arms, identical seeds, identical episode set; only the instruction strings differ between them.

    pip install provael==0.44.0
    provael run --policy smolvla --suite libero --task-suite libero_object --attack none --seeds 5 --out baseline/
    provael run --policy smolvla --suite libero --task-suite libero_object --attack paraphrase --seeds 5 --out arm/
    provael report --format markdown baseline/ arm/

Three things to hold fixed, or the comparison says nothing. Same seeds in both arms, because the policy sampler is a draw, and two runs of one checkpoint are two draws. Same episode set. And report the benign arm's number beside the other one, never on its own.

If you run it, the per-attack table carries a Wilson interval per arm, and at 50 episodes those intervals are wide enough that adjacent arms often overlap. That is the honest picture, not a flaw in the run.

One ask, and it is the reason I am offering the runner rather than just the argument. The reproduction register on my side is empty. If you do run both arms, would you be willing to post the two numbers in this thread? Whatever they say, including nothing.

**Comment by atirna** [comment]
dont have a gpu-backed runner to hand, so both arms would be cpu inference and 100 episodes of smolvla at that speed is not a realistic turnaround for me. numbers from a handicapped runner would be worse than none, so my register stays empty.

the wiring half holds up from the rename map itself. the instruction-string control is yours to run if you want it, and ill take the numbers either way.
